commercetools/types_store.go

```diff
@@ -144,153 +188,155 @@ func (obj StoreReference) MarshalJSON() ([]byte, error) {
+	}{Action: "removeDistributionChannel", Alias: (*Alias)(&obj)})
 }
 
-// StoreSetLanguagesAction implements the interface StoreUpdateAction
-type StoreSetLanguagesAction struct {
-	Languages []string `json:"languages,omitempty"`
+// StoreRemoveSupplyChannelAction implements the interface StoreUpdateAction
+type StoreRemoveSupplyChannelAction struct {
+	SupplyChannel *ChannelResourceIdentifier `json:"supplyChannel"`
 }
 
 // MarshalJSON override to set the discriminator value
-func (obj StoreSetLanguagesAction) MarshalJSON() ([]byte, error) {
-	type Alias StoreSetLanguagesAction
+func (obj StoreRemoveSupplyChannelAction) MarshalJSON() ([]byte, error) {
+	type Alias StoreRemoveSupplyChannelAction
 	return json.Marshal(struct {
 		Action string `json:"action"`
 		*Alias
-	}{Action: "setLanguages", Alias: (*Alias)(&obj)})
+	}{Action: "removeSupplyChannel", Alias: (*Alias)(&obj)})
 }
 
-// StoreSetNameAction implements the interface StoreUpdateAction
-type StoreSetNameAction struct {
-	Name *LocalizedString `json:"name,omitempty"`
+// StoreResourceIdentifier implements the interface ResourceIdentifier
+type StoreResourceIdentifier struct {
+	Key string `json:"key,omitempty"`
+	ID  string `json:"id,omitempty"`
 }
 
 // MarshalJSON override to set the discriminator value
-func (obj StoreSetNameAction) MarshalJSON() ([]byte, error) {
-	type Alias StoreSetNameAction
+func (obj StoreResourceIdentifier) MarshalJSON() ([]byte, error) {
+	type Alias StoreResourceIdentifier
 	return json.Marshal(struct {
-		Action string `json:"action"`
+		TypeID string `json:"typeId"`
 		*Alias
-	}{Action: "setName", Alias: (*Alias)(&obj)})
-}
-
-// StoreUpdate is a standalone struct
-type StoreUpdate struct {
-	Version int                 `json:"version"`
-	Actions []StoreUpdateAction `json:"actions"`
-}
-
-// UnmarshalJSON override to deserialize correct attribute types based
-// on the discriminator value
-func (obj *StoreUpdate) UnmarshalJSON(data []byte) error {
-	type Alias StoreUpdate
-	if err := json.Unmarshal(data, (*Alias)(obj)); err != nil {
-		return err
-	}
-	for i := range obj.Actions {
-		var err error
-		obj.Actions[i], err = mapDiscriminatorStoreUpdateAction(obj.Actions[i])
-		if err != nil {
-			return err
-		}
-	}
-
-	return nil
+	}{TypeID: "store", Alias: (*Alias)(&obj)})
 }
 
-// StoresAddDistributionChannelsAction implements the interface StoreUpdateAction
-type StoresAddDistributionChannelsAction struct {
-	DistributionChannel *ChannelResourceIdentifier `json:"distributionChannel"`
+// StoreSetCustomFieldAction implements the interface StoreUpdateAction
+type StoreSetCustomFieldAction struct {
+	Value interface{} `json:"value,omitempty"`
+	Name  string      `json:"name"`
 }
 
 // MarshalJSON override to set the discriminator value
-func (obj StoresAddDistributionChannelsAction) MarshalJSON() ([]byte, error) {
-	type Alias StoresAddDistributionChannelsAction
+func (obj StoreSetCustomFieldAction) MarshalJSON() ([]byte, error) {
```

Comment:
Is the `Stores` to `Store` change for several of these actions a commercetools change or caused by a codegen change?

Reply:
I couldn't pinpoint how this is generated but indeed, it could be related to the codegen change. It seems that all the `Stores` Actions have been renamed to `Store`.

I listed all the breaking change in the changelog: https://github.com/labd/commercetools-go-sdk/pull/59/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed

Reply:
Awesome! LGTM. So I'll go ahead and merge